nucypher/network/nodes.py

```diff
@@ -973,26 +984,40 @@ def update_snapshot(self, checksum, updated, number_of_known_nodes):
     # Stamp
     #
 
-    def _stamp_has_valid_wallet_signature(self) -> bool:
-        """Off-chain Signature Verification of ethereum client signature of stamp"""
+    def _stamp_has_valid_signature_by_worker(self) -> bool:
```

Comment:
Why is this method on "Teacher"?
Isn't Teacher primarily a learning loop designation?
I feel like all these signature/stamp validation methods should be on a power or on some other character base class called "Validator" or something

Reply:
@vepkenez That's a great comment. In our character-based model, nodes create a local representation of other remote nodes via "Stranger" characters (i.e., `self.is_me is False`); the cool thing about this model is that even if it's not the real, remote node, you can still treat it as such and ask it many things, including to validate itself. The way I see it is that `Teacher` represents a "character" from which you learn something, which includes information about the validity of its identity.